packages/@react-spectrum/combobox/src/MobileComboBox.tsx

```diff
@@ -83,6 +84,15 @@ export const MobileComboBox = React.forwardRef(function MobileComboBox<T extends
     }
   };
 
+  // When mobile ComboBox button recieves focus, set state.isFocused (i.e. the tray input's focus tracker) to false.
+  // This is to prevent state.isFocused from being set to true when the tray closes
+  // (FocusScope attempts to restore focus to the tray input when tapping outside the tray due to "contain")
```

Comment:
Interesting. Where is that focusing of the button happening? Doesn't FocusScope handle that via `restoreFocus`? And at that point it will have unmounted so contain shouldn't apply anymore? If not, then that's a problem in FocusScope that might affect other components.

Reply:
To reproduce, you can change https://github.com/adobe/react-spectrum/blob/main/packages/@react-spectrum/overlays/src/Overlay.tsx#L40 to be `!isOpen` so that the overlay immediately disappears, making it so that https://github.com/adobe/react-spectrum/blob/main/packages/@react-aria/focus/src/FocusScope.tsx#L244 will not fire since the overlay and its contents are unmounted

Reply:
What first comes to mind is informing the FocusScope that an element/the FocusScope itself is transitioning out/being removed and thus suspend focus restoring/containment for the element/the FocusScope itself

Reply:
Hmm but the focus restoration happens in a requestAnimationFrame after the focus scope has unmounted. By that point the events should be unbound? https://github.com/adobe/react-spectrum/blob/main/packages/@react-aria/focus/src/FocusScope.tsx#L378

Reply:
Just to make sure we are on the same page, this is the flow that currently happens from what I can tell:
1. User taps on underlay to close the tray. 
2. This blurs the combobox tray input. 
3. FocusScope's contain behavior refocuses the tray input via https://github.com/adobe/react-spectrum/blob/main/packages/@react-aria/focus/src/FocusScope.tsx#L235-L247. This triggers because the FocusScope hasn't unmounted at this point due to Overlay's transition time. This means combobox's `state.isFocused` is true.
4. The tray closes and the FocusScope unmounts. The FocusScope's containment behavior focus focuses the ComboBox button via https://github.com/adobe/react-spectrum/blob/main/packages/@react-aria/focus/src/FocusScope.tsx#L378 but `state.isFocused` is still true hence why I have the change here to set that to false.

Am I misunderstanding something in the above?

Reply:
Oh, so onBlur doesn't fire when the tray input unmounts?

Reply:
yep, thats right, looks like react doesn't call onBlur on unmount: https://github.com/facebook/react/issues/12363

Reply:
Can we handle the underlying problem instead? i.e. set state.isFocused to false when the tray unmounts rather than handling focus on the button, which is a side effect of the former.

Reply:
Ah, derp, makes sense. I'll make it a useEffect cleanup in the ComboBox tray component